dt_model/symbols/constraint.py

```diff
@@ -13,7 +32,13 @@ class Constraint:
     This class is used to define constraints for the model.
     """
 
-    def __init__(self, usage: Symbol, capacity: Symbol, group: str | None = None, name: str = "") -> None:
+    def __init__(
+        self,
+        usage: graph.Node,
+        capacity: graph.Node | CumulativeDistribution,
+        group: str | None = None,
+        name: str = "",
+    ) -> None:
```

Comment:
Consider making the old `Constraint` class private and re-exposing it only for legacy purposes. That said, I am no fully convinced this is the best possible structure for the code, hence need to second guess myself.

Reply:
The issue here is that the original code had an underlying `.value` and the constraint did not contain a `graph.Node` or a `Distribution`, rather it contained something that contained a symbol or a distribution.

It remains to be seen if I want to adapt the code to go down this road or this is fine. It's probably fine but it causes some DX issues when writing the model, so maybe it's not so good.

Reply:
OK, I tried to implement another approach here.